python/tk_multi_launchapp/software_entity_launcher.py

```diff
@@ -79,17 +68,25 @@ def register_launch_commands(self):
                     )
                     continue
 
-            app_display_name = sw_entity["code"]
-            app_args = sw_entity[app_args_field] or ""
-            register_cmd_data.extend(
-                self._build_register_command_data(
-                    app_display_name,
-                    local_thumb_path,
-                    app_engine,
-                    app_path,
-                    app_args,
-                    app_versions
+            # If a thumbnail was registered for this Software entity, download it
+            # and/or use the locally cached file instead.
+            if app_icon and self._tk_app.engine.has_ui:
+                # import sgutils locally as this has dependencies on QT
+                shotgun_data = sgtk.platform.import_framework(
+                    "tk-framework-shotgunutils", "shotgun_data"
+                )
+                # download thumbnail from shotgun and cache for reuse.
+                self._tk_app.log_debug("Download app icon...")
```

Comment:
if i am reading the code correctly, it looks like the thumbnail is always downloaded if it exists - regardless of whether it is later being used or not. Since the download is a slow operation, do you think we could do this lazily, so that we only download the icon if we absolutely have to? I suspect this approach would potentially speed things up for people trying to launch desktop!